common/types/src/gateway.rs

```diff
@@ -75,3 +76,39 @@ impl GatewayBond {
         })
     }
 }
+
+#[derive(Serialize, Deserialize)]
+pub struct GatewayNodeDetailsResponse {
+    pub identity_key: String,
+    pub sphinx_key: String,
+    pub owner_signature: String,
+    pub announce_address: String,
+    pub bind_address: String,
+    pub version: String,
+    pub mix_port: u16,
+    pub clients_port: u16,
+    pub data_store: String,
+}
+
+impl fmt::Display for GatewayNodeDetailsResponse {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        writeln!(f, "Identity Key: {}", self.identity_key).expect("Could not write line");
```

Comment:
Since it's a `Display` trait, why not just propagate the error rather than using `expect`, i.e. `writeln!(f, "Identity Key: {}", self.identity_key)?`